src/Doctrine/Orm/Filter/IriSearchFilter.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Doctrine\Orm\Filter;
+
+use ApiPlatform\Doctrine\Common\Filter\IriSearchFilterTrait;
+use ApiPlatform\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Metadata\OpenApiParameterFilterInterface;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\Metadata\ParameterProviderFilterInterface;
+use ApiPlatform\Metadata\PropertiesFilterInterface;
+use ApiPlatform\State\Provider\IriConverterParameterProvider;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\QueryBuilder;
+use Doctrine\Persistence\ManagerRegistry;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+
+class IriSearchFilter extends AbstractFilter implements OpenApiParameterFilterInterface, PropertiesFilterInterface, ParameterProviderFilterInterface
+{
+    use IriSearchFilterTrait;
+
+    public function __construct(
+        ?ManagerRegistry $managerRegistry = null,
+        ?LoggerInterface $logger = null,
+        ?array $properties = null,
+        ?NameConverterInterface $nameConverter = null,
+    ) {
+        parent::__construct($managerRegistry, $logger, $properties, $nameConverter);
+    }
+
+    protected function filterProperty(string $property, mixed $value, QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, ?Operation $operation = null, array $context = []): void
+    {
+        if (
+            null === $value
+            || !$this->isPropertyEnabled($property, $resourceClass)
+            || !$this->isPropertyMapped($property, $resourceClass, true)
+        ) {
+            return;
+        }
+
+        $extraProperties = $operation?->getExtraProperties();
+        $resource = $extraProperties['_value'] ?? null;
+        if (!$resource) {
+            $this->logger->warning(\sprintf('No resource found for property "%s".', $property));
+
+            return;
+        }
+
+        $alias = $queryBuilder->getRootAliases()[0];
+        $parameterName = $queryNameGenerator->generateParameterName($property);
+
+        $queryBuilder
+            ->andWhere(\sprintf('%s.%s = :%s', $alias, $property, $parameterName))
+            ->setParameter($parameterName, $resource->getId());
```

Comment:
Properties should be available inside the `$context['parameter']`:

If the parameter has `:property` you should get an array of properties inside `Parameter::extraProperties` if not use `Parameter::property` and in the end the `Parameter::key`.

The only thing we need to do here is to check if our doctrine entity has that property/association (re-use some of the code that we already have for that).